src/libbootimg.c

```diff
@@ -92,6 +92,9 @@ void libbootimg_init_new(struct bootimg *img)
     memset(img, 0, sizeof(struct bootimg));
     memcpy(img->hdr.magic, BOOT_MAGIC, BOOT_MAGIC_SIZE);
     img->hdr.page_size = DEFAULT_PAGE_SIZE;
+    img->hdr_elf = malloc(sizeof(struct boot_img_hdr_elf));
```

Comment:
Any specific reason why this is allocated on heap instead of being part of the struct bootimg? Doesn't seem that big.